src/service/compact/merge.rs

```diff
@@ -375,8 +423,17 @@ pub async fn merge_by_stream(
                 }
             }
             let mut worker_results = Vec::with_capacity(batch_group_len);
+            let mut updated_at = config::utils::time::now_micros();
             for _ in 0..batch_group_len {
                 let result = inner_rx.recv().await.unwrap();
+                let now = config::utils::time::now_micros();
+                // convert job_timeout from secs to micros, and check 1/4 of job_timeout
+                if now - updated_at > (cfg.compact.job_run_timeout * 1000 * 1000 / 4) {
+                    if let Err(e) = infra_file_list::update_running_jobs(job_id).await {
+                        log::error!("[COMPACT] update_running_jobs failed: {e}");
+                    }
+                    updated_at = now;
+                }
```

Comment:
is it possible that this `if` block is always skipped? then this `job_id` would never be updated, but later `infra_file_list::set_job_done(job_id)` will set it to be done. in that case, an additional clause `where status = Pending` in `infra_file_list::set_job_done(job_id)` would be helpful.

Reply:
added comment